### PR TITLE
Add documentation for setting up multiple iCub robots

### PR DESCRIPTION
**Pull request by Yeshasvitvs**
This PR adds the documentation for setting up multiple iCub robots either in simulation or real hardware on the same network.

ezgif com-gif-maker (8)

CC @lrapetti @danielepucci

### CODE REVIEW

**Review by pattacini — CHANGES_REQUESTED** [review]
First off, thanks @Yeshasvitvs a lot for the addition 👍🏻 

This documentation is not simply markdown-based but rather it's in mkdocs and as such we need to comply with some constraints.

For example, new pages are required to be linked to the "nav" section as per the following warning (accessible through the CI):
```
INFO    -  The following pages exist in the docs directory, but are not included in the "nav" configuration:
  - icub_setup/index.md
  - icub_setup/how_to_use_two_icub_robots_in_gazebo_simulation.md
  - icub_setup/how_to_use_two_real_iCub_robots_on_the_same_network.md 
```

This means that you ought to edit the `mkdocs.yml` file as well.

I would rename `icub_setup` in something more specific like `icub_setup_multiple_robots` and, also, the name of the files contained therein are really too long. Better off something like the following:
- `two_robots_simulation.md`
- `two_robots_real.md`

Would you mind addressing these points?

That said, it would really much easier to work out and edit all this within Gitpod, as per the instructions in the main README. You'll have an environment ready for you; just type in the console `mkdocs serve` and you'll get the results.

**Review by pattacini — APPROVED** [review]

File: docs/icub_setup_multiple_robots/two_robots_simulation.md [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+
+
+If you are using the two repositories [icub-gazebo](https://github.com/robotology/icub-gazebo) and [icub-gazebo-wholebody](https://github.com/robotology/icub-gazebo-wholebody) as a part of [robotology-superbuild](https://github.com/robotology/robotology-superbuild) you need to build the superbuild and make sure the changes made to [gazebo_icub_robotname.ini](https://github.com/robotology/icub-gazebo/blob/master/icub/conf/gazebo_icub_robotname.ini) configuration file are correctly reflected in the `install` directory.
+
+Now, when a first iCub model is spawned in gazebo it takes the default name of `iCub` and all the ports related to it are prefixed with `/iCub`. Similarly, an additional iCub model spawned next to it will have the default name `iCub_0` and all the ports related to this new model are prefixed with `/iCub_0`. In this way multiple robots can be spawned in gazebo simulation environment without any port conflicts.
+
+<img src ="../img/two_icubs_gazebo.png.png" width=400>
```
Comment (Yeshasvitvs) [inline docs/icub_setup_multiple_robots/two_robots_simulation.md:21]: @pattacini `.png` extension is wrong, to be fixed

File: docs/icub_setup_multiple_robots/two_robots_simulation.md [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+Another important detail concerns the use of `wholeBodyDynamics`. In general, `wholeBodyDynamics` is launched with `yarprobotinterface` with a configuration file like launch-wholebodydynamics.xml that uses `/icubSim` port prefix as shown below:
+
+<img src ="../img/wholebodynamics_example.png.png" width=400>
```
Comment (Yeshasvitvs) [inline docs/icub_setup_multiple_robots/two_robots_simulation.md:25]: @pattacini `.png` extension is wrong, to be fixed

**Comment by Yeshasvitvs** [comment]
Screenshot from 2020-10-15 15-20-17

The figure extensions are wrong

File: docs/icub_setup_multiple_robots/two_robots_simulation.md (hunk 1 above)
Comment (pattacini) [reply docs/icub_setup_multiple_robots/two_robots_simulation.md:21]: Already done 👍🏻

File: docs/icub_setup_multiple_robots/two_robots_simulation.md (hunk 2 above)
Comment (pattacini) [reply docs/icub_setup_multiple_robots/two_robots_simulation.md:25]: Done

**Pull request merged**